corehq/apps/reports/filters/api.py

```diff
@@ -298,21 +266,28 @@ def user_es_query(self, query):
 
 
 @location_safe
-class LocationRestrictedEmwfOptions(LocationRestrictedEmwfOptionsMixin, EmwfOptionsView):
-    def extra_data_sources(self):
-        return []
-
-
-@location_safe
-class CaseListFilterOptions(LocationRestrictedEmwfOptionsMixin, EmwfOptionsView):
+class CaseListFilterOptions(EmwfOptionsView):
 
     @property
     @memoized
     def utils(self):
         return CaseListFilterUtils(self.domain)
 
-    def extra_data_sources(self):
-        return [(self.get_sharing_groups_size, self.get_sharing_groups)]
+    @property
+    def data_sources(self):
+        if self.request.can_access_all_locations:
+            return [
+                (self.get_static_options_size, self.get_static_options),
+                (self.get_groups_size, self.get_groups),
+                (self.get_sharing_groups_size, self.get_sharing_groups),
+                (self.get_locations_size, self.get_locations),
+                (self.get_users_size, self.get_users),
+            ]
+        else:
+            return [
+                (self.get_locations_size, self.get_locations),
+                (self.get_users_size, self.get_users),
+            ]
```

Comment:
Might want to still retain `extra_data_sources` to avoid repeating this.

Reply:
Just commented on the above one, but I prefer to lay it out all in one place, so you can easily tell which sources are being used.  Otherwise you'd have to jump to the super class to and then figure out what's being added or removed from that to determine which sources this uses.  This is kinda like "The case list filter uses these ones".

Reply:
Although I just realized that this will need to account for the special locations query too...

Reply:
@mkangia check out my latest two commits - I made it so the ES queries don't fail when special characters are provided, which means we can remove the restriction for special locations queries.  We _do_ want users to be able to search for users using quotes too, I think.